File: v1/scalatra/src/main/scala/com/wordnik/client/api/PostsApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.Void

import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class PostsApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "PostsApi"
  override protected val applicationName: Option[String] = Some("Posts")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }
  

  val indexOperation = (apiOperation[Void]("index")
      summary "Read a list of posts"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        queryParam[String]("filter_by").description("").optional
        
        
        
        
        
        ,
        queryParam[String]("filter_by_group_id").description("").optional
        
        
        
        
        
        ,
        queryParam[String]("filter_by_newsletter_id").description("").optional
        
        
        
        
        
        
        )
  )

  get("api/advisors/{advisor_id}/posts.json",operation(indexOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    

    
      
      val filter_by = params.getAs[String]("filter_by")
      
    

    

    

    
    
    println("filter_by: " + filter_by)
  
    
    
    

    
      
      val filter_by_group_id = params.getAs[String]("filter_by_group_id")
      
    

    

    

    
    
    println("filter_by_group_id: " + filter_by_group_id)
  
    
    
    

    
      
      val filter_by_newsletter_id = params.getAs[String]("filter_by_newsletter_id")
      
    

    

    

    
    
    println("filter_by_newsletter_id: " + filter_by_newsletter_id)
  
  }

  

  val createOperation = (apiOperation[Void]("create")
      summary "Create an post"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        
        
        
        formParam[String]("post").description("")
        
        
        )
  )

  post("api/advisors/{advisor_id}/posts.json",operation(createOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    

    

    

    
      val post = params.getAs[String]("post")
    

    
    
    println("post: " + post)
  
  }

  

  val showOperation = (apiOperation[Void]("show")
      summary "Read a specific post"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        
        )
  )

  get("api/advisors/{advisor_id}/posts/{id}.json",operation(showOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
  }

  

  val updateOperation = (apiOperation[Void]("update")
      summary "Update a post"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        
        
        
        formParam[String]("post").description("")
        
        
        )
  )

  put("api/advisors/{advisor_id}/posts/{id}.json",operation(updateOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    

    

    

    
      val post = params.getAs[String]("post")
    

    
    
    println("post: " + post)
  
  }

  

  val destroyOperation = (apiOperation[Void]("destroy")
      summary "Deletes a post"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        
        )
  )

  delete("api/advisors/{advisor_id}/posts/{id}.json",operation(destroyOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
  }

}

File: v1/nodejs/app/apis/PostsApi.js
var swagger = require("swagger-node-express");
var url = require("url");
var errors = swagger.errors;
var params = swagger.params;

/* add model includes */

function writeResponse (response, data) {
  response.header('Access-Control-Allow-Origin', "*");
  response.header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
  response.header("Access-Control-Allow-Headers", "Content-Type");
  response.header("Content-Type", "application/json; charset=utf-8");
  response.send(JSON.stringify(data));
}

exports.models = models = require("../models.js");

exports.index = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/posts.json",
    "notes" : "This retrieves a list of post objects",
    "summary" : "Read a list of posts",
    "method": "GET",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    
      params.query("filter_by", "Supports [all/published/unpublished/features/vestorly/newsletter]", "", false, false, ""),
    
      params.query("filter_by_group_id", "Filter by group id", "", false, false, ""),
    
      params.query("filter_by_newsletter_id", "Filter by newsletter", "", false, false, "")
    ].concat([]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "index"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing index as a GET method?"});    
  }
};
exports.create = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/posts.json",
    "notes" : "This creates an post in the database",
    "summary" : "Create an post",
    "method": "POST",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, "")
    ].concat([
      params.path("advisor_id", "Id of advisor")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "create"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing create as a POST method?"});    
  }
};
exports.show = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/posts/{id}.json",
    "notes" : "This retrieves a specific post object",
    "summary" : "Read a specific post",
    "method": "GET",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, "")
    ].concat([
      params.path("id", "Id of post"),
    
      params.path("advisor_id", "Id of advisor")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "show"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing show as a GET method?"});    
  }
};
exports.update = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/posts/{id}.json",
    "notes" : "This updates a post in the database",
    "summary" : "Update a post",
    "method": "PUT",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, "")
    ].concat([
      params.path("id", "Id of post"),
    
      params.path("advisor_id", "Id of advisor")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "update"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing update as a PUT method?"});    
  }
};
exports.destroy = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/posts/{id}.json",
    "notes" : "This deletes a post in the database",
    "summary" : "Deletes a post",
    "method": "DELETE",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, "")
    ].concat([
      params.path("id", "Id of advisor")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "destroy"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing destroy as a DELETE method?"});    
  }
};
